python/ql/src/semmle/python/objects/ObjectAPI.qll

```diff
@@ -0,0 +1,261 @@
+/**
+ * Public API for "objects"
+ * A `Value` is a static approximation to a set of runtime objects.
+ */
+
+
+
+
+import python
+private import TObject
+private import semmle.python.objects.ObjectInternal
+private import semmle.python.pointsto.PointsTo
+private import semmle.python.pointsto.PointsToContext
+
+/* Use the term `ObjectSource` to refer to DB entity. Either a CFG node
+ * for Python objects, or `@py_cobject` entity for built-in objects.
+ */
+class ObjectSource = Object;
+
+/** Class representing values in the Python program
+ * Each `Value` is a static approximation to a set of one or more real objects.
+ */
+class Value extends TObject {
+
+    Value() {
+        this != ObjectInternal::unknown() and
+        this != ObjectInternal::unknownClass() and
+        this != ObjectInternal::undefined()
+    }
+
+    string toString() {
+        result = this.(ObjectInternal).toString()
+    }
+
+    /** Gets a `ControlFlowNode` that refers to this object. */
+    ControlFlowNode getAReference() {
+        PointsToInternal::pointsTo(result, _, this, _)
+    }
+
+    /** Gets the class of this object.
+     * Strictly, the `Value` representing the class of the objects
+     * represented by this Value.
+     */
+    Value getClass() {
+        result = this.(ObjectInternal).getClass()
+    }
+
+    /** Gets a call to this object */
+    CallNode getACall() {
+        PointsToInternal::pointsTo(result.getFunction(), _, this, _)
```

Comment:
It looks to me like this body could be replaced with simply `result = this.getACall(_)`.